version 2.2.12 -- Call to undefined method `   Rector\PhpParser\Printer\BetterStandardPrinter::pInterpolatedStringPart()`

rector version 2.2.13,  php8.4, debian13

```❯ ./vendor/bin/rector --version
Rector 2.2.12
```

```
[ERROR] Could not process                                                                                      
         "/data/Work/DOGNOSIS/mudhol.dognosis.link/apps/ehr-portal/app/Controllers/ApiHtmx.php" file, due to:   
         "System error: "Call to undefined method                                                               
         Rector\PhpParser\Printer\BetterStandardPrinter::pInterpolatedStringPart()"                             
                                                                                                                
         Stack trace:                                                                                           
         #0 vendor/rector/rector/src/PhpParser/Printer/BetterStandardPrinter.php(127):                          
         PhpParser\PrettyPrinterAbstract->p()                                                                   
         #1 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(357):          
         Rector\PhpParser\Printer\BetterStandardPrinter->p()                                                    
         #2 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(272):          
         PhpParser\PrettyPrinterAbstract->pStmts()                                                              
         #3 vendor/rector/rector/src/PhpParser/Printer/BetterStandardPrinter.php(91):                           
         PhpParser\PrettyPrinterAbstract->prettyPrint()                                                         
         #4 vendor/rector/rector/src/PhpParser/Comparing/NodeComparator.php(31):                                
         Rector\PhpParser\Printer\BetterStandardPrinter->print()                                                
         #5 vendor/rector/rector/src/PhpParser/Comparing/NodeComparator.php(50):                                
         Rector\PhpParser\Comparing\NodeComparator->printWithoutComments()                                      
         #6 vendor/rector/rector/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php(110):                
         Rector\PhpParser\Comparing\NodeComparator->areNodesEqual()                                             
         #7 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/FirstFindingVisitor.php(42): 
         Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector->{closure:Rector\DeadCode\Rector\Assign\RemoveD 
         oubleAssignRector::isSelfReferencing():110}()                                                          
         #8 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(171):                  
         PhpParser\NodeVisitor\FirstFindingVisitor->enterNode()                                                 
         #9 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(90):                   
         PhpParser\NodeTraverser->traverseArray()                                                               
         #10 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(198):                 
         PhpParser\NodeTraverser->traverseNode()                                                                
         #11 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(71):                  
         PhpParser\NodeTraverser->traverseArray()                                                               
         #12 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeFinder.php(64):                     
         PhpParser\NodeTraverser->traverse()                                                                    
         #13 vendor/rector/rector/src/PhpParser/Node/BetterNodeFinder.php(151):                                 
         PhpParser\NodeFinder->findFirst()                                                                      
         #14 vendor/rector/rector/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php(110):               
         Rector\PhpParser\Node\BetterNodeFinder->findFirst()                                                    
         #15 vendor/rector/rector/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php(82):                
         Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector->isSelfReferencing()                            
         #16 vendor/rector/rector/src/Rector/AbstractRector.php(115):                                           
         Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector->refactor()                                     
         #17 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(160):          
         Rector\Rector\AbstractRector->enterNode()                                                              
         #18 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78):           
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray()                         
         #19 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187):          
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode()                          
         #20 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78):           
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray()                         
         #21 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187):          
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode()                          
         #22 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(60):           
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray()                         
         #23 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(51):                      
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverse()                              
         #24 vendor/rector/rector/src/Application/FileProcessor.php(95):                                        
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse()                                         
         #25 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(178):                            
         Rector\Application\FileProcessor->processFile()                                                        
         #26 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(152):                            
         Rector\Application\ApplicationFileProcessor->processFile()                                             
         #27 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(128):                            
         Rector\Application\ApplicationFileProcessor->processFiles()                                            
         #28 vendor/rector/rector/src/Console/Command/ProcessCommand.php(172):                                  
         Rector\Application\ApplicationFileProcessor->run()                                                     
         #29 vendor/rector/rector/vendor/symfony/console/Command/Command.php(289):                              
         Rector\Console\Command\ProcessCommand->execute()                                                       
         #30 vendor/rector/rector/vendor/symfony/console/Application.php(899):                                  
         RectorPrefix202512\Symfony\Component\Console\Command\Command->run()                                    
         #31 vendor/rector/rector/vendor/symfony/console/Application.php(279):                                  
         RectorPrefix202512\Symfony\Component\Console\Application->doRunCommand()                               
         #32 vendor/rector/rector/src/Console/ConsoleApplication.php(60):                                       
         RectorPrefix202512\Symfony\Component\Console\Application->doRun()                                      
         #33 vendor/rector/rector/vendor/symfony/console/Application.php(162):                                  
         Rector\Console\ConsoleApplication->doRun()                                                             
         #34 vendor/rector/rector/bin/rector.php(130):                                                          
         RectorPrefix202512\Symfony\Component\Console\Application->run()                                        
         #35 vendor/rector/rector/bin/rector(5): require_once('...')                                            
         #36 vendor/bin/rector(119): include('...')                                                             
         #37 {main}". On line: 579                                                                              
                                                                                                                

                                                                                                                
 [ERROR] Could not process                                                                                      
         "/data/Work/DOGNOSIS/mudhol.dognosis.link/apps/ehr-portal/app/Views/admin/manages_locations.php" file, 
         due to:                                                                                                
         "System error: "Call to undefined method                                                               
         Rector\PhpParser\Printer\BetterStandardPrinter::pInterpolatedStringPart()"                             
                                                                                                                
         Stack trace:                                                                                           
         #0 vendor/rector/rector/src/PhpParser/Printer/BetterStandardPrinter.php(127):                          
         PhpParser\PrettyPrinterAbstract->p()                                                                   
         #1 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(357):          
         Rector\PhpParser\Printer\BetterStandardPrinter->p()                                                    
         #2 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(272):          
         PhpParser\PrettyPrinterAbstract->pStmts()                                                              
         #3 vendor/rector/rector/src/PhpParser/Printer/BetterStandardPrinter.php(91):                           
         PhpParser\PrettyPrinterAbstract->prettyPrint()                                                         
         #4 vendor/rector/rector/src/PhpParser/Comparing/NodeComparator.php(31):                                
         Rector\PhpParser\Printer\BetterStandardPrinter->print()                                                
         #5 vendor/rector/rector/src/PhpParser/Comparing/NodeComparator.php(50):                                
         Rector\PhpParser\Comparing\NodeComparator->printWithoutComments()                                      
         #6 vendor/rector/rector/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php(110):                
         Rector\PhpParser\Comparing\NodeComparator->areNodesEqual()                                             
         #7 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/FirstFindingVisitor.php(42): 
         Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector->{closure:Rector\DeadCode\Rector\Assign\RemoveD 
         oubleAssignRector::isSelfReferencing():110}()                                                          
         #8 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(171):                  
         PhpParser\NodeVisitor\FirstFindingVisitor->enterNode()                                                 
         #9 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(90):                   
         PhpParser\NodeTraverser->traverseArray()                                                               
         #10 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(198):                 
         PhpParser\NodeTraverser->traverseNode()                                                                
         #11 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(71):                  
         PhpParser\NodeTraverser->traverseArray()                                                               
         #12 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeFinder.php(64):                     
         PhpParser\NodeTraverser->traverse()                                                                    
         #13 vendor/rector/rector/src/PhpParser/Node/BetterNodeFinder.php(151):                                 
         PhpParser\NodeFinder->findFirst()                                                                      
         #14 vendor/rector/rector/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php(110):               
         Rector\PhpParser\Node\BetterNodeFinder->findFirst()                                                    
         #15 vendor/rector/rector/rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php(82):                
         Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector->isSelfReferencing()                            
         #16 vendor/rector/rector/src/Rector/AbstractRector.php(115):                                           
         Rector\DeadCode\Rector\Assign\RemoveDoubleAssignRector->refactor()                                     
         #17 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(160):          
         Rector\Rector\AbstractRector->enterNode()                                                              
         #18 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78):           
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray()                         
         #19 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187):          
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode()                          
         #20 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(78):           
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray()                         
         #21 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(187):          
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseNode()                          
         #22 vendor/rector/rector/src/PhpParser/NodeTraverser/AbstractImmutableNodeTraverser.php(60):           
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverseArray()                         
         #23 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(51):                      
         Rector\PhpParser\NodeTraverser\AbstractImmutableNodeTraverser->traverse()                              
         #24 vendor/rector/rector/src/Application/FileProcessor.php(95):                                        
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse()                                         
         #25 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(178):                            
         Rector\Application\FileProcessor->processFile()                                                        
         #26 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(152):                            
         Rector\Application\ApplicationFileProcessor->processFile()                                             
         #27 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(128):                            
         Rector\Application\ApplicationFileProcessor->processFiles()                                            
         #28 vendor/rector/rector/src/Console/Command/ProcessCommand.php(172):                                  
         Rector\Application\ApplicationFileProcessor->run()                                                     
         #29 vendor/rector/rector/vendor/symfony/console/Command/Command.php(289):                              
         Rector\Console\Command\ProcessCommand->execute()                                                       
         #30 vendor/rector/rector/vendor/symfony/console/Application.php(899):                                  
         RectorPrefix202512\Symfony\Component\Console\Command\Command->run()                                    
         #31 vendor/rector/rector/vendor/symfony/console/Application.php(279):                                  
         RectorPrefix202512\Symfony\Component\Console\Application->doRunCommand()                               
         #32 vendor/rector/rector/src/Console/ConsoleApplication.php(60):                                       
         RectorPrefix202512\Symfony\Component\Console\Application->doRun()                                      
         #33 vendor/rector/rector/vendor/symfony/console/Application.php(162):                                  
         Rector\Console\ConsoleApplication->doRun()                                                             
         #34 vendor/rector/rector/bin/rector.php(130):                                                          
         RectorPrefix202512\Symfony\Component\Console\Application->run()                                        
         #35 vendor/rector/rector/bin/rector(5): require_once('...')                                            
         #36 vendor/bin/rector(119): include('...')                                                             
         #37 {main}". On line: 579                                                                              
                                                                                                          
```

--- a/rules-tests/DeadCode/Rector/Assign/RemoveDoubleAssignRector/Fixture/compare_interpolated_string.php.inc
+++ b/rules-tests/DeadCode/Rector/Assign/RemoveDoubleAssignRector/Fixture/compare_interpolated_string.php.inc
@@ -1,0 +1,22 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\Assign\RemoveDoubleAssignRector\Fixture;
+
+final class CompareInterpolatedString
+{
+    public function run()
+    {
+        $markdown = 'a long string';
+
+        $markdown = preg_replace_callback(
+            function (array $matches): string {
+                $match = $matches[1];
+                $className = strtolower(Toolkit::toSlug($match));
+
+                return "<span class='tag tag-{$className}'>{$match}</span>";
+            },
+            (string)$markdown
+        );
+
+    }
+}

--- a/src/PhpParser/Printer/BetterStandardPrinter.php
+++ b/src/PhpParser/Printer/BetterStandardPrinter.php
@@ -21,6 +21,7 @@ use PhpParser\Node\Expr\Match_;
 use PhpParser\Node\Expr\MethodCall;
 use PhpParser\Node\Expr\Ternary;
 use PhpParser\Node\Expr\Yield_;
+use PhpParser\Node\InterpolatedStringPart;
 use PhpParser\Node\Scalar\InterpolatedString;
 use PhpParser\Node\Scalar\String_;
 use PhpParser\Node\Stmt\Declare_;
@@ -110,6 +111,15 @@ final class BetterStandardPrinter extends Standard
     public function pFileWithoutNamespace(FileWithoutNamespace $fileWithoutNamespace): string
     {
         return $this->pStmts($fileWithoutNamespace->stmts);
+    }
+
+    /**
+     * Use for standalone InterpolatedStringPart printing, that is not support by php-parser natively.
+     * Used e.g. in \Rector\PhpParser\Comparing\NodeComparator::printWithoutComments
+     */
+    protected function pInterpolatedStringPart(InterpolatedStringPart $interpolatedStringPart): string
+    {
+        return $interpolatedStringPart->value;
     }
 
     protected function p(

